### PR TITLE
[FAU-376] Synchronize child terms for degree and admission requirement taxonomies

### PR DESCRIPTION
**Pull request by zhyian**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature

**What is the current behavior?** (You can also link to an open issue here)
https://inpsyde.atlassian.net/browse/FAU-376

**What is the new behavior (if this is a feature change)?**
Child terms are synchronized for the Degree and Admission Requirement taxonomies. Additionally, slugs for the Admission Requirement taxonomies are updated to support the pre-filtered feature.

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

**Other information**:
Psaml issues should be fixed after merge https://github.com/RRZE-Webteam/FAU-Studium-Common/pull/15

### CODE REVIEW

File: src/Infrastructure/Search/FilterableTermsUpdater.php [hunk 1]
```diff
@@ -222,48 +222,177 @@ private function maybeCreateTerms(
...
+     * @return int
+     */
+    private function retrieveParentTermId(
+        MultilingualString|MultilingualLink|AdmissionRequirement|Degree $flatProperty,
+        array &$taxonomyCache
+    ): int {
+
+        if (!$flatProperty instanceof Degree && !$flatProperty instanceof AdmissionRequirement) {
+            return 0;
+        }
```
Comment (shvlv) [inline src/Infrastructure/Search/FilterableTermsUpdater.php:340]: Does extracting the condition in something like `isHierarchicalPropery` make sense?

**Review by shvlv — APPROVED** [review]
Thanks!

File: src/Infrastructure/Search/TermData.php [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    public function slug(): string
+    {
+        return $this->slug;
+    }
+
+    public function termId(): int
+    {
+
```
Comment (o-samaras) [inline src/Infrastructure/Search/TermData.php:53]:
```suggestion
```

**Review by o-samaras — APPROVED** [review]
LGTM. Thanks

**Pull request merged**